x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/terms/index.tsx

```diff
@@ -62,16 +63,20 @@ function ofName(name?: string, count: number = 0) {
   });
 }
 
+function isScriptedField(field: IndexPatternField): boolean;
```

Comment:
Is the overload necessary here? It seems like the last definition is capturing everything.

Reply:
Initially I had only the generic third signature, but then thought it may be useful to express to the consumer why an `indexPattern` was required even when a field object was passed.
So the idea of the separate signatures which strongly express the relationship between the string name version with the `indexPattern`, compared to the other one.

Not particular strong opinion here, as the implementation is actually the same, just a type enhancement.

Reply:
ah got it, good point 👍